src/Search/Extensions/ProxyDBExtension.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace SilverStripe\FullTextSearch\Search\Extensions;
+
+use SilverStripe\Core\Extension;
+use TractorCow\ClassProxy\Generators\ProxyGenerator;
+use SilverStripe\FullTextSearch\Search\Updaters\SearchUpdater;
+
+class ProxyDBExtension extends Extension
+{
+    public function updateProxy(ProxyGenerator &$proxy)
```

Comment:
Some doc blocks would be good to explain what this class is doing

Reply:
```
/**
 * Class ProxyDBExtension
 * @package SilverStripe\FullTextSearch\Search\Extensions
 *
 * Proxies the database by decorating the factory with code to be extended.
 * Methods can be chained; all addMethod() calls on the same method name will form a set of middleware.
 * First methods registered are executed first.
 *
 */
class ProxyDBExtension extends Extension
{
    /**
     * @param ProxyGenerator $proxy
     */
    public function updateProxy(ProxyGenerator &$proxy)
```
... is a bit to high level, is it?

Reply:
I guess so. Devs could get this from the proxy db module.


The point of these manipulateion calls in the context of fulltextsearch is to "Check every index to see what objects need re-inserting into what indexes to keep the index fresh, but doesn't actually do it yet" (see `SearchUpdater::handle_manipulation`.

Perhaps we could summarise a combination of the two... something like "This database connector proxy will allow {@link SearchUpdater::handle_manipulation} to monitor database schema changes that may need to be propagated through to search indexes.".

On a side note, from reading this code I can't actually see where this binds itself to the `Database` class. I see where you're registering the callback for the `manipulate` method, but presumably this doesn't run for every `manipulate` method call in the entire system right? Is there something missing?

Reply:
@tractorcow Want to help me out with this one, please? I like Robbie's suggestion for the class doc block. When it comes to the updateProxy() one though I find it hard to summarise what I don't fully understand 😉

Reply:
Thanks team for your feedback and input!